File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/abstract_factory/lab/organic_garden.cpp
#include "organic_garden.h"

organic_garden::organic_garden(organic_lettuce* a_lettuce_prototype)
    : garden(a_lettuce_prototype)
{
}

void organic_garden::set_lettuce_prototype(organic_lettuce* a_lettuce_prototype)
{
    garden::set_lettuce_prototype(a_lettuce_prototype);
}

organic_tomato* organic_garden::make_tomato() const
{
	return new organic_tomato(3, 1);
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/singleton/abstract_factory/lab/OrganicGarden.java
// OrganicGarden.java

public class OrganicGarden extends Garden
{
	public OrganicGarden(OrganicLettuce aLettucePrototype)
	{
		super(aLettucePrototype);
	}

	public void setLettucePrototype(OrganicLettuce aLettucePrototype)
	{
		super.setLettucePrototype(aLettucePrototype);
	}

	public OrganicTomato makeTomato()
	{
		return new OrganicTomato(1, 2);
	}
}
